src/dialogs/create-file.tsx

```diff
@@ -0,0 +1,233 @@
+                      {_("Cancel")}
+                  </Button>
+              </>
+          }
+        >
+            <Stack>
+                {create_error &&
+                <Alert
+                  className="file-editor-alert"
+                  variant="danger"
+                  title={create_error}
+                  isInline
+                />}
+                <Form isHorizontal>
+                    <FormGroup
+                      label={_("File name")}
+                      fieldId="file-name"
+                    >
+                        <TextInput
+                          autoFocus // eslint-disable-line jsx-a11y/no-autofocus
+                          id="file-name"
+                          isRequired
+                          value={filename}
+                          onChange={(_event, value) => {
+                              setFileNameError(checkFilename(value, cwdInfo?.entries || {}, undefined));
+                              setFilename(value);
+                          }}
+                        />
+                        <FormHelper fieldId="file-name" helperTextInvalid={filename_error} />
+                    </FormGroup>
+                    <TextArea
+                      id="editor-text-area"
+                      className="file-editor"
+                      value={initial_text}
+                      onChange={(_ev, content) => setInitialText(content)}
+                    />
+                    {candidates.length > 0 &&
+                        <FormGroup fieldId="create-file-owner" label={_("Owner")}>
+                            <FormSelect
+                              id='create-file-owner'
+                              value={owner}
+                              onChange={(_ev, val) => setOwner(val)}
+                            >
+                                {candidates.map(owner =>
+                                    <FormSelectOption
+                                      key={owner}
+                                      value={owner}
+                                      label={owner}
+                                    />)}
+                            </FormSelect>
+                        </FormGroup>}
+                </Form>
+            </Stack>
+        </Modal>
+    );
+};
+
+export async function show_create_file_dialog(dialogs: Dialogs, path: string,
+    addAlert: (title: string, variant: AlertVariant, key: string, detail?: string) => void) {
+    if (!superuser.allowed) {
+        try {
+            await cockpit.spawn(["test", "-w", path]);
+        } catch {
+            addAlert(_("Cannot create file in current directory"), AlertVariant.warning, "create-file",
+                     _("Permission denied"));
+            return;
+        }
+    }
+    const filename = await dialogs.run(CreateFileModal, { path });
+    if (filename !== "") {
+        edit_file(dialogs, filename);
+    }
```

Comment:
What does this do? This looks like creation and edit dialogs would be two separate things. But the CreateFileModal has its own "initial text" field.  (screenshot in the description shows that as well). The commit message and the test are trying to tell me something, but I don't understand it.